httpcore/_backends/asyncio.py

```diff
@@ -0,0 +1,228 @@
+import asyncio
+import socket
+import ssl
+from typing import Any, Dict, Iterable, Optional, Type
+
+from .._exceptions import (
+    ConnectError,
+    ConnectTimeout,
+    ReadError,
+    ReadTimeout,
+    WriteError,
+    WriteTimeout,
+    map_exceptions,
+)
+from .._utils import is_socket_readable
+from .base import SOCKET_OPTION, AsyncNetworkBackend, AsyncNetworkStream
+
+
+class AsyncIOStream(AsyncNetworkStream):
+    def __init__(
+        self, stream_reader: asyncio.StreamReader, stream_writer: asyncio.StreamWriter
+    ):
+        self._stream_reader = stream_reader
+        self._stream_writer = stream_writer
+        self._read_lock = asyncio.Lock()
+        self._write_lock = asyncio.Lock()
```

Comment:
We don't use locking at the stream layer, let's remove these.

(I think we had locking at this layer in the past, the locking is now handled at the `HTTP11Connection`/`HTTP2Connection` layer)

Reply:
Done, they are now gone